### ISSUE TITLE
a suggest

### ISSUE THREAD

**Issue by lucktu** [ntop/n2n#38]
edgenode ---> edge, so
supernode ---> super?
It's very nice to see your back !

**Comment by emanuele-f** [comment]
Well, I agree that names are a little weird, but look at microsoft "edge", "edge" computing, edge is the future :) . No plans for changes for now.

**Comment by xmxsuperstar** [comment]
i like:
edge -> n2n edge
super -> n2n super

**Comment by emanuele-f** [comment]
they contain spaces and that's not allowed for program names

**Comment by lucktu** [comment]
Other people’s main program and software name are basically the same(zerotier/tinc/frp... ...), n2n is not, and It’s not easy to spread, so I want to change the name again, launched a vote.

https://xoyondo.com/ap/fJf7Lg48Vz3PEKu

**Comment by Logan007** [comment]
For the very soon upcoming 3.x series, there currently are no plans to change the well-known name scheme of _edge_ and _supernode_. Especially as  the whole documentation and support in the _Issue_'s discussions use these terms coined by all previous versions. A lot of users are used to them.

From where I see it, the next opportunity to chose names is when we merge the executlables into one, probably with 4.0. But then, the natural choice is likely to be `n2n`?

**Comment by lucktu** [comment]
I agree with you that merging the executlables into one (n2n), future.

Of course, if not very troublesome, early change also has the advantage of early change, easy publicity, leaving more relics

**Comment by xiahuaijia** [comment]
Merge into one executable file n2n is easy to spread.
But I don't like using numbers in the file name on the command line. Does anyone have a better idea?

super -> `n2n -center`
edge -> `n2n -node`

**Comment by Logan007** [comment]
The merger idea is not about merging current executables. It is more about merging function: Future versions (4.x) will probably use a slightly changed protocol so every node in the network can act as edge and / or supernode.

The basic idea is that all nodes are equal and interconnect more like the distributed peer-to-peer networks (see the other discussion over at #405). Then, there will be no need to specify `-center` or `-node` because their behavior will just depend on their position in the network and if an outlet (virtual TAP) is defined at that specific node.
